File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.py
import math

def solution(arr):
    lcm = int(arr[0] * arr[1] / math.gcd(arr[0],arr[1]))
    for i in range(2,len(arr)):
        lcm = int(lcm * arr[i] / math.gcd(lcm, arr[i]))
    return lcm

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
const gcd = (a, b) => {
    return b ? gcd(b, a%b) : a
}

function solution(arr) {
    let lcm = arr.shift()
    
    arr.sort((a, b) => a - b)
    
    while(arr.length !== 0) {
        const val = arr.shift()
        lcm = lcm * val / gcd(val, lcm)
    }
    
    return lcm
}